pxr/usd/lib/usd/stage.cpp

```diff
@@ -3154,12 +3154,25 @@ UsdStage::_DefinePrim(const SdfPath &path, const TfToken &typeName)
 {
     // Special-case requests for the root.  It always succeeds and never does
     // authoring since the root cannot have PrimSpecs.
-    if (path == SdfPath::AbsoluteRootPath())
+    if (path == SdfPath::AbsoluteRootPath()) {
         return GetPseudoRoot();
+    }
 
     // Define all ancestors.
-    if (!_DefinePrim(path.GetParentPath(), TfToken()))
+    auto parentPath = path.GetParentPath();
+    if (!_DefinePrim(parentPath, TfToken())) {
         return UsdPrim();
+    }
+
+    // Disallow authoring a prim under an inactive ancestor
```

Comment:
So this fixes UsdStage::DefinePrim, but UsdStage::OverridePrim has the same issue (even though it doesn't issue the same runtime error that DefinePrim does).

I think this logic may instead belong in UsdStage::_ValidateEditPrimAtPath, which gets called by UsdStage::_IsValidPathForCreatingPrim, which gets called by both DefinePrim and OverridePrim. There's a separate UsdStage::_ValidateEditPrim function, but I don't believe we need this logic there -- that function gets a valid UsdPrim, which means that prim must already be active. 

If we move this logic into _ValidateEditPrimAtPath, we should check that it's not called anywhere that might be a tight loop -- a cursory look suggest we'd be OK. Also, a coding error would be appropriate instead of a warning in that function.